app/assets/javascripts/app/models/post/interactions.js

```diff
@@ -103,23 +103,21 @@ app.models.Post.Interactions = Backbone.Model.extend({
       , reshare = this.post.reshare()
       , flash = new Diaspora.Widgets.FlashMessages();
 
-    reshare.save({}, {
-      success : function(){
+    reshare.save()
+      .done(function(){
         flash.render({
           success: true,
           notice: Diaspora.I18n.t("reshares.successful")
         });
-      },
-      error: function(){
+        interactions.reshares.add(reshare);
+        if (app.stream) {app.stream.addNow(reshare)};
```

Comment:
No you didn't. `git rebase -i upstream/develop` should open an editor where you can choose `pick`, `squash`, `fix` and so on. You keep the first commit at `pick` and choose `squash` for the others.